templates/commands/render.go

```diff
@@ -322,6 +343,92 @@ func (r *Render) realRun(ctx context.Context, rp *runParams) (outErr error) {
 	return nil
 }
 
+// processUnknownInputs checks for any unknown input flags and returns an error.
+func processUnknownInputs(flagInputs map[string]string, spec *model.Spec) error {
+	var merr error
+
+	specInputs := make(map[string]any)
+	for _, v := range spec.Inputs {
+		specInputs[v.Name.Val] = struct{}{}
+	}
+
+	for key, value := range flagInputs {
+		if _, ok := specInputs[key]; !ok {
+			merr = errors.Join(merr, fmt.Errorf("  - %s=%s", key, value))
+		}
+	}
+
+	return merr
+}
+
+// validateFlagInputs validates the input values.
+func (r *Render) validateFlagInputs(spec *model.Spec) error {
+	var merr error
+
+	for _, i := range spec.Inputs {
+		_, ok := r.flagInputs[i.Name.Val]
+		if !ok {
+			if !r.flagQuiet {
+				if err := r.promptForInput(i); err != nil {
+					return err
+				}
+			} else {
+				if i.Default != nil {
+					r.flagInputs[i.Name.Val] = i.Default.Val
+				} else {
+					merr = errors.Join(merr, fmt.Errorf("  - %s", i.Name.Val))
+				}
+			}
+		}
+	}
+
+	return merr
+}
+
+// promptForInput prompts the user for a value for a missing input. The prompt will continue
+// until a value is provided unless the input was configured with a default value, it will be
+// used for an empty input provided by the user.
+func (r *Render) promptForInput(i *model.Input) error {
```

Comment:
Not sure this is the best way to do this but.......open to feedback.

Reply:
https://github.com/abcxyz/pkg/blob/main/cli/command.go#L277